### ISSUE TITLE
Is it safe to re-add a lost backend?

### ISSUE THREAD

**Issue by nlowe** [NetApp/trident#184]
Trident v18.07 in Kubernetes 1.10.3

When I [tried to deploy trident to a separate kubernetes cluster](https://github.com/NetApp/trident/issues/175#issuecomment-422496342) (with the same NetApp instance) I did not change the `igroupName` or `storagePrefix`. I believe this overwrote or corrupted some of the configuration for my other cluster that was already deployed. I noticed today that the trident pod in my main cluster was stuck in a crash loop. It was complaining it couldn't find the backend for one (and only one) of my volumes. I `exec`'d into the `etcd` pod and something is definitely wrong here. There were only two keys:

```bash
/ # ETCDCTL_API=3 etcdctl --endpoints=http://127.0.0.1:8001 get --prefix /
/trident/store
{"store_version":"etcdv3","orchestrator_api_version":"1"}
/trident/v1/volume/myvol-1b7c1
...
```

I'm guessing this isn't supposed to look like this given that my other cluster has an entry for the backend and all of the PV's deployed. If I delete the `myvol` key the pod starts successfully.

Here's my question: Is it safe to `tridentctl create` the backend again? Will trident re-discover all of my pv's?

**Comment by adkerr** [comment]
If you installed trident into the same SVM and didn't change the trident volume name then it would have detected the previous ETCD volume and attempted to use it (as it would assuming this was a reinstall or upgrade). Though usually this should have resulted in an ETCD locking problem and the ETCD container would not have started (assuming the older trident was running at the time).

Typically Trident will not auto-discover volumes on the backend, even if they match the prefix, as it keeps an internal record of all the volumes it knows it created. I'm not clear on the state of your ETCD database, so I hesitate to say what will happen if you re-add the backend to Trident.

@kangarlou probably is the best person to answer your specific question, but be aware that he's at NetApp Insight this week and may be slow to respond.

You may also try opening an official support case with NetApp so we can get you a more timely response. 
http://mysupport.netapp.com/info/web/ECMLP2619434.html

**Comment by jacobjohnanda** [comment]
I agree with Andrew. When we try to mount the same volume for another Trident setup on a different Kubernetes cluster,it ends up in a ETCD locking issue and the second installation will not go through and the pod will not get instantiated.

**Comment by nlowe** [comment]
Hmm, OK, maybe that's not what happened then. I'll keep digging. Thanks for the input!
